sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/InferFiltersFromConstraintsSuite.scala

```diff
@@ -192,4 +192,17 @@ class InferFiltersFromConstraintsSuite extends PlanTest {
 
     comparePlans(Optimize.execute(original.analyze), correct.analyze)
   }
+
+  test("SPARK-23405:single left-semi join, filter out nulls on either side on equi-join keys") {
```

Comment:
nit: `SPARK-23405: left-semi equa-join should filter out null join keys on both sides`